service/src/main/java/uk/gov/hmcts/reform/fpl/service/removeorder/RemoveOrderService.java

```diff
@@ -0,0 +1,82 @@
+package uk.gov.hmcts.reform.fpl.service.removeorder;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.common.dynamic.DynamicList;
+import uk.gov.hmcts.reform.fpl.model.interfaces.RemovableOrder;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.asDynamicList;
+
+@Service
+@RequiredArgsConstructor(onConstructor_ = @Autowired)
+public class RemoveOrderService {
+
+    private final OrderRemovalActions orderRemovalActions;
+
+    @SuppressWarnings("unchecked")
+    public DynamicList buildDynamicListOfOrders(CaseData caseData, UUID selected) {
+        List<Element<RemovableOrder>> blankOrders = getRemovableOrderList(caseData).stream()
+            .filter(order -> order.getValue().isRemovable())
+            .map(order -> (Element<RemovableOrder>) order)
+            .collect(Collectors.toList());
+
+        return asDynamicList(blankOrders, selected, RemovableOrder::asLabel);
+    }
+
+    public DynamicList buildDynamicListOfOrders(CaseData caseData) {
+        return buildDynamicListOfOrders(caseData, null);
+    }
+
+    public void populateSelectedOrderFields(CaseData caseData,
+                                            Map<String, Object> data,
+                                            UUID removedOrderId,
+                                            RemovableOrder removableOrder) {
+        orderRemovalActions
+            .getActions()
+            .stream()
+            .filter(orderRemovalAction -> orderRemovalAction.isAccepted(removableOrder))
+            .findFirst()
+            .orElseThrow(() -> new IllegalArgumentException(
+                format("Action not found for order %s", removedOrderId)))
```

Comment:
I feel like this part can be moved to a method in the `OrderRemovalActions` class. There can be a method called getAction that does this so that this service only has to worry about calling the method on the action instead of getting the right one.